sbin/submitty_autograding_shipper.py

```diff
@@ -488,51 +517,110 @@ def unpack_job(which_machine, which_untrusted, next_directory, next_to_grade, ra
             # Close file descriptors
             for var in [fd1, fd2]:
                 if var:
-                    try:
+                    with contextlib.suppress(OSError):
                         os.close(var)
-                    except Exception:
-                        pass
 
-        if not success:
-            return False
+        # If we are waiting, return so now. Otherwise, we can check if
+        # the job returned successfully
+        if status in [GradingStatus.WAITING, GradingStatus.FAILURE]:
+            return status
 
     try:
         with open(local_done_queue_file, 'r') as infile:
             local_done_queue_obj = json.load(infile)
+
         # Check to make certain that the job we received was the correct job
         if random_identifier != local_done_queue_obj['identifier']:
-            success = False
+            status = GradingStatus.WAITING
             msg = f"{which_machine} returned a stale job (ids don't match). Discarding."
+            # Even though this job was not the one we are waiting for, report errors.
+            if local_done_queue_obj['autograding_status']['status'] == 'fail':
+                msg += ' discarded job failed. Check the stack traces log for details.'
+                autograding_utils.log_stack_trace(
+                    AUTOGRADING_STACKTRACE_PATH, job_id=JOB_ID,
+                    trace=f"ERROR: {worker_name} returned the following error for a stale job:\n"
+                          f"{local_done_queue_obj['autograding_status']['message']}"
+                )
+            print(msg)
```

Comment:
stray print?

Reply:
Not enough prints, actually. We want the shipper to be a little bit chatty on stdout when it receives a completed job, in case it is being run manually (sometimes done for debugging). This one survived my refactor, but a few others were taken out. I added them back in a recent commit.